From c4d7c9dca97955ab0e93c516568b5c52667e9570 Mon Sep 17 00:00:00 2001
From: Antonio Terceiro <terceiro@debian.org>
Date: Wed, 19 Aug 2015 22:54:18 +0200
Subject: [PATCH] lxc-debian: allow not including contrib/non-free

Signed-off-by: Antonio Terceiro <terceiro@debian.org>
---
 templates/lxc-debian.in | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

--- a/templates/lxc-debian.in
+++ b/templates/lxc-debian.in
@@ -162,13 +162,19 @@ write_sourceslist()
         prefix="deb [arch=${arch}]"
     fi
 
+    if [ "$mainonly" = 1 ]; then
+      non_main=''
+    else
+      non_main=' contrib non-free'
+    fi
+
     cat >> "${rootfs}/etc/apt/sources.list" << EOF
-${prefix} $MIRROR          ${release}         main contrib non-free
+${prefix} $MIRROR          ${release}         main${non_main}
 EOF
 
     if [ "$release" != "unstable" -a "$release" != "sid" ]; then
       cat >> "${rootfs}/etc/apt/sources.list" << EOF
-${prefix} $SECURITY_MIRROR ${release}/updates main contrib non-free
+${prefix} $SECURITY_MIRROR ${release}/updates main${non_main}
 EOF
     fi
 }
@@ -455,7 +461,8 @@ Template specific options can be passed
   lxc-create --name=NAME [-lxc-create-options] -- [-template-options]
 
 Usage: $1 -h|--help -p|--path=<path> [-c|--clean] [-a|--arch=<arch>] [-r|--release=<release>]
-                                     [--mirror=<mirror>] [--security-mirror=<security mirror>]
+                                     [--main-only] [--mirror=<mirror>]
+                                     [--security-mirror=<security mirror>]
 
 Options :
 
@@ -465,6 +472,7 @@ Options :
                          amd64, armhf, armel, powerpc. Defaults to host arch.
   -r, --release=RELEASE  Debian release. Can be one of: squeeze, wheezy, jessie, sid.
                          Defaults to current stable.
+  --main-only            Include only Debian's main repository (no contrib/non-free)
   --mirror=MIRROR        Debian mirror to use during installation. Overrides the MIRROR
                          environment variable (see below).
   --security-mirror=SECURITY_MIRROR
@@ -483,7 +491,7 @@ EOF
     return 0
 }
 
-options=$(getopt -o hp:n:a:r:c -l arch:,clean,help,mirror:,name:,path:,release:,rootfs:,security-mirror: -- "$@")
+options=$(getopt -o hp:n:a:r:c -l arch:,clean,help,main-only,mirror:,name:,path:,release:,rootfs:,security-mirror: -- "$@")
 if [ $? -ne 0 ]; then
         usage $(basename $0)
         exit 1
@@ -508,6 +516,7 @@ do
 
         -a|--arch)            arch=$2; shift 2;;
         -c|--clean)           clean=1; shift 1;;
+           --main-only)       mainonly=1; shift 1;;
            --mirror)          MIRROR=$2; shift 2;;
         -n|--name)            name=$2; shift 2;;
         -p|--path)            path=$2; shift 2;;
