Author: Christian Seiler <christian@iwakd.de>
Description: Fix FTBFS on ia64 (clone vs. __clone2)

--- a/src/tests/reboot.c
+++ b/src/tests/reboot.c
@@ -45,11 +45,16 @@ static int do_reboot(void *arg)
 static int test_reboot(int cmd, int sig)
 {
         long stack_size = 4096;
-        void *stack = alloca(stack_size) + stack_size;
+        void *stack = alloca(stack_size);
         int status;
         pid_t ret;
 
+#if defined(__ia64__)
+        ret = __clone2(do_reboot, stack, stack_size, CLONE_NEWPID | SIGCHLD,  &cmd);
+#else
+        stack += stack_size;
         ret = clone(do_reboot, stack, CLONE_NEWPID | SIGCHLD, &cmd);
+#endif
         if (ret < 0) {
                 printf("failed to clone: %m\n");
                 return -1;
